### PR TITLE
Mbm 34 select spells

### PR DESCRIPTION
**Pull request by insertStringHere**

### CODE REVIEW

**Review by solidwaterslayer — APPROVED** [review]

**Comment by MattKirchh** [comment]
Are we ok allowing the user to be able to right-click use a spell from the hotbar? Overriding the `use` function allows for that behavior.

**Comment by MattKirchh** [comment]
The left-hand magic select screen can be opened even if no spells are equipped, giving a blank screen with no UI elements.

File: src/main/java/com/mineboundteam/minebound/magic/network/MagicSync.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    public static void registerPackets(FMLCommonSetupEvent event) {
+        NET_CHANNEL.registerMessage(0, ButtonMsg.class, ButtonMsg::encode, ButtonMsg::decode,
+                MagicSync::handleButtons);
+    }
+
+    public static void handleButtons(ButtonMsg msg, Supplier<NetworkEvent.Context> ctx) {
+        ctx.get().enqueueWork(() -> {
+            ServerPlayer player =  ctx.get().getSender();
+            if(!player.level.isClientSide()){
+                ItemStack chest = player.getItemBySlot(EquipmentSlot.CHEST);
```
Comment (MattKirchh) [inline src/main/java/com/mineboundteam/minebound/magic/network/MagicSync.java:41]: How come we are only checking the chestpiece here and no other armor?

**Comment by MattKirchh** [comment]
Since we are overriding the `use` function to trigger the spells, how would I go about getting the `ItemStack` of the spell being used? Since it's wont necessarily be held in hand by the player.

**Comment by MattKirchh** [comment]
> Since we are overriding the `use` function to trigger the spells, how would I go about getting the `ItemStack` of the spell being used? Since it's wont necessarily be held in hand by the player.

Would it be `getItem` in InventorySpellContainer?

File: src/main/java/com/mineboundteam/minebound/magic/network/MagicSync.java (hunk 1 above)
Comment (insertStringHere) [reply src/main/java/com/mineboundteam/minebound/magic/network/MagicSync.java:41]: only the chestpiece has the hand slots on it; if you're not wearing a chestpiece then you won't have access to the spell library

**Comment by insertStringHere** [comment]
> Since we are overriding the `use` function to trigger the spells, how would I go about getting the `ItemStack` of the spell being used? Since it's wont necessarily be held in hand by the player.

from the hotbar? I have no idea; from the suit, the capability points to an armor piece and the slot within that piece. Those slots store itemstacks

File: src/main/java/com/mineboundteam/minebound/magic/network/MagicSync.java (hunk 1 above)
Comment (MattKirchh) [reply src/main/java/com/mineboundteam/minebound/magic/network/MagicSync.java:41]: So even if the player has active spells equipped on other pieces, they won't be able to select and use them unless they have a Myrial chestpiece equipped?

Comment (insertStringHere) [reply src/main/java/com/mineboundteam/minebound/magic/network/MagicSync.java:41]: Yeah, I think we also have a bauble that gives that capability

**Comment by MattKirchh** [comment]
> > Since we are overriding the `use` function to trigger the spells, how would I go about getting the `ItemStack` of the spell being used? Since it's wont necessarily be held in hand by the player.
> 
> from the hotbar? I have no idea; from the suit, the capability points to an armor piece and the slot within that piece. Those slots store itemstacks

When it's used from the hotbar we can just use the `InteractionHand` parameter to get it. I assume that means we're ok allowing the player to activate spells from the hotbar.

**Comment by MattKirchh** [comment]
> > > Since we are overriding the `use` function to trigger the spells, how would I go about getting the `ItemStack` of the spell being used? Since it's wont necessarily be held in hand by the player.
> > 
> > 
> > from the hotbar? I have no idea; from the suit, the capability points to an armor piece and the slot within that piece. Those slots store itemstacks
> 
> When it's used from the hotbar we can just use the `InteractionHand` parameter of `use` to get it. I assume that means we're ok allowing the player to activate spells from the hotbar.

**Comment by insertStringHere** [comment]
> > > Since we are overriding the `use` function to trigger the spells, how would I go about getting the `ItemStack` of the spell being used? Since it's wont necessarily be held in hand by the player.
> > 
> > 
> > from the hotbar? I have no idea; from the suit, the capability points to an armor piece and the slot within that piece. Those slots store itemstacks
> 
> When it's used from the hotbar we can just use the `InteractionHand` parameter of `use` to get it. I assume that means we're ok allowing the player to activate spells from the hotbar.

Yeah, that's been the intent from the start I think; I believe we mentioned it in the design doc

**Review by MattKirchh — APPROVED** [review]

**Review by MattKirchh — APPROVED** [review]

**Pull request merged**